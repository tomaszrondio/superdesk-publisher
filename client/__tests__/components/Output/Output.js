import React from "react";
import Output from "../../../components/Output/Output";
import { render, wait } from "@testing-library/react";
import Publisher from "../../../__mocks__/publisher";

const publisher = new Publisher();

jest.mock("react-virtualized/styles.css", () => {
  return {};
});
jest.mock("axios");
jest.mock("react-select", () => props => "div");

let api = () => { };

api.users = {};
api.users.query = function () {
  return new Promise((resolve, reject) => {
    resolve({
      _items: [{ is_author: true, display_name: "test author" }],
      _links: {}
    });
  });
};

const languages = [
  { qcode: 'pl', name: 'Polish' },
  { qcode: 'en', name: 'English' },
  { qcode: 'de', name: 'German' },
]

describe("Output/Output", () => {
  it("renders properly", async () => {
    const { container, getByText } = render(
      <Output
        publisher={publisher}
        notify={{ error: jest.fn() }}
        config={{}}
        authoringWorkspace={{}}
        api={api}
        isLanguagesEnabled={true}
        languages={languages}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
