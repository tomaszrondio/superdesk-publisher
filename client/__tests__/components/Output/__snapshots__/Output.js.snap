// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/Output renders properly 1`] = `
<div
  class="sd-page-content__content-block sd-page-content__content-block--double-sidebar"
>
  <div
    class="subnav"
  >
    <button
      class="navbtn navbtn--left navbtn--darker"
      flow="right"
      sd-tooltip="Filter"
    >
      <i
        class="icon-filter-large"
      />
    </button>
    <h3
      class="subnav__page-title sd-flex-no-grow"
    >
      Output Control
    </h3>
    <div
      class="flat-searchbar search-handler--left-border"
    >
      <div
        class="search-handler search-handler--left-border"
      >
        <label
          class="trigger-icon"
          for="search-input"
        >
          <i
            class="icon-search"
          />
        </label>
        <input
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="subnav subnav--lower-z-index"
  >
    <div
      class="subnav__content-bar"
    >
      <span
        class="sd-check__wrapper"
      >
        <span
          class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio checked"
        >
          <label>
            Incoming content
          </label>
        </span>
      </span>
      <span
        class="sd-check__wrapper"
      >
        <span
          class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio"
        >
          <label>
            Published
          </label>
        </span>
      </span>
    </div>
    <div
      class="subnav__content-bar sd-flex-wrap ml-auto sd-padding-l--1"
    >
      <div
        class="dropdown dropdown--align-right"
        flow="left"
      >
        <span>
          <button
            class="dropdown__toggle"
          >
            Updated
            <span
              class="dropdown__caret"
            />
          </button>
        </span>
        <ul
          class="dropdown__menu dropdown--align-right"
        >
          <li>
            <button>
              Updated
            </button>
          </li>
          <li>
            <button>
              Created
            </button>
          </li>
        </ul>
      </div>
      <a
        class="icn-btn"
        flow="bottom"
        sd-tooltip="Descending"
      >
        <i
          class="icon-descending"
        />
      </a>
    </div>
    <div
      class="subnav__spacer subnav__spacer--no-margin"
    />
    <div
      class="subnav__content-bar sd-flex-wrap sd-padding-l--1 ui-responsive-small--d-none ui-responsive-small--d-block if-authoring--d-none"
    >
      <div
        class="sd-margin-r--1"
      >
        <span>
          Incoming:
        </span>
        <span
          class="badge sd-margin-l--0-5"
        >
          0
        </span>
      </div>
      <div>
        <span>
          Published:
        </span>
        <span
          class="badge sd-margin-l--0-5"
        >
          0
        </span>
      </div>
    </div>
    <div
      class="subnav__spacer subnav__spacer--no-margin"
    />
    <div
      class="subnav__content-bar sd-flex-no-shrink sd-margin-x--0"
    >
      <div
        class="dropdown dropdown--align-right"
      >
        <span>
          <button
            class="dropdown__toggle navbtn navbtn--text-only dropdown-toggle"
          >
            All Languages
            <span
              class="dropdown__caret"
            />
          </button>
        </span>
        <ul
          class="dropdown__menu dropdown__menu--scrollable"
        >
          <li>
            <button>
              All Languages
            </button>
          </li>
          <li
            class="dropdown__menu-divider"
          />
          <li>
            <button>
              Polish
            </button>
          </li>
          <li>
            <button>
              English
            </button>
          </li>
          <li>
            <button>
              German
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="sd-column-box--3"
  >
    <div
      class="sd-filters-panel sd-filters-panel--border-right"
    >
      <div
        class="side-panel side-panel--transparent side-panel--shadow-right"
      >
        <div
          class="side-panel__header side-panel__header--border-b"
        >
          <a
            class="icn-btn side-panel__close"
            flow="left"
            sd-tooltip="Close filters"
          >
            <i
              class="icon-close-small"
            />
          </a>
          <h3
            class="side-panel__heading side-panel__heading--big"
          >
            Advanced Filter
          </h3>
        </div>
        <div
          class="side-panel__content"
        >
          <div
            class="side-panel__content-block"
          >
            <div
              class="form__row"
            >
              <div
                class="sd-line-input sd-line-input--no-margin sd-line-input--with-button"
              >
                <label
                  class="sd-line-input__label"
                >
                  Routes
                </label>
                div
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="sd-line-input sd-line-input--no-margin sd-line-input--with-button"
              >
                <label
                  class="sd-line-input__label"
                >
                  Authors
                </label>
                div
              </div>
            </div>
            <div
              class="form__row form__row--flex"
            >
              <div
                class="sd-line-input sd-line-input--no-margin"
              >
                <label
                  class="sd-line-input__label"
                >
                  Published after
                </label>
                <input
                  class="sd-line-input__input"
                  name="published_after"
                  type="date"
                  value=""
                />
              </div>
            </div>
            <div
              class="form__row form__row--flex"
            >
              <div
                class="sd-line-input sd-line-input--no-margin"
              >
                <label
                  class="sd-line-input__label"
                >
                  Published before
                </label>
                <input
                  class="sd-line-input__input"
                  name="published_before"
                  type="date"
                  value=""
                />
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="sd-line-input sd-line-input--is-select"
              >
                <label
                  class="sd-line-input__label"
                >
                  Ingest source
                </label>
                div
              </div>
            </div>
          </div>
        </div>
        <div
          class="side-panel__footer side-panel__footer--button-box"
        >
          <div
            class="flex-grid flex-grid--boxed-small flex-grid--small-2"
          >
            <a
              class="btn btn--hollow"
            >
              Clear
            </a>
            <a
              class="btn btn--primary"
            >
              Filter
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sd-column-box__main-column relative"
    >
      <div
        class="sd-loader"
        style="z-index: 1;"
      />
    </div>
  </div>
</div>
`;
